mono/metadata/threads.c

```diff
@@ -2012,11 +2015,13 @@ gint32 ves_icall_System_Threading_WaitHandle_WaitOne_internal(HANDLE handle, gin
 	if (mono_thread_current_check_pending_interrupt ())
 		return map_native_wait_result_to_managed (MONO_W32HANDLE_WAIT_RET_FAILED, 0);
 
+	mono_w32handle_ref(handle);
```

Comment:
Scenario (1) what this PR implements for the WaitHandle.WaitAll/WaitAny icalls (see locking `scan_mutex` in `mono_w32handle_load_from_monoarray`. I will add this to the single-waithandle icalls too, but the problem has yet to show up here (the multiple-waithandles path takes a much longer time to get to inc'ing the ref).